Test/Utils/ContrailAPI/ConfigureDNS.Tests.ps1

```diff
@@ -0,0 +1,132 @@
+            -AuthUrl $OpenStackConfig.AuthUrl() `
+            -Username $OpenStackConfig.Username `
+            -Password $OpenStackConfig.Password `
+            -Tenant $OpenStackConfig.Project
+    }
+
+    It 'adds and removes "empty" contrail DNS server' {
+        $ServerUUID = Add-ContrailDNSServer -ContrailUrl $ContrailUrl -AuthToken $AuthToken `
+            -DNSServerName "CreatedByPS1Script"
+        Remove-ContrailDNSServer -ContrailUrl $ContrailUrl -AuthToken $AuthToken -DNSServerUuid $ServerUUID
+    }
+
+    It 'adds and removes DNS server records' {
+        $ServerUUID = Add-ContrailDNSServer -ContrailUrl $ContrailUrl -AuthToken $AuthToken `
+            -DNSServerName "CreatedByPS1Script"
+
+        $Record1Data = [VirtualDNSRecordData]::new("host1", "1.2.3.4", "A")
+        $Record1UUID = Add-ContrailDNSRecord -ContrailUrl $ContrailUrl -AuthToken $AuthToken `
+            -DNSServerName "CreatedByPS1Script" -VirtualDNSRecordData $Record1Data
+
+        $Record2Data = [VirtualDNSRecordData]::new("host2", "1.2.3.5", "A")
+        $Record2UUID = Add-ContrailDNSRecord -ContrailUrl $ContrailUrl -AuthToken $AuthToken `
+            -DNSServerName "CreatedByPS1Script" -VirtualDNSRecordData $Record2Data
+
+        Remove-ContrailDNSRecord -ContrailUrl $ContrailUrl -AuthToken $AuthToken -DNSRecordUuid $Record1UUID
+        Remove-ContrailDNSRecord -ContrailUrl $ContrailUrl -AuthToken $AuthToken -DNSRecordUuid $Record2UUID
+
+        Remove-ContrailDNSServer -ContrailUrl $ContrailUrl -AuthToken $AuthToken -DNSServerUuid $ServerUUID
+    }
+
+    It 'breaks when wrong dns mode specified' {
+        { Set-IpamDNSMode -ContrailUrl $ContrailUrl -AuthToken $AuthToken `
+            -IpamFQName @("default-domain", "default-project", "default-network-ipam") `
+            -DNSMode 'wrongdnsmode' } | Should -Throw
+    }
+
+    It 'sets DNS mode to none' {
+        Set-IpamDNSMode -ContrailUrl $ContrailUrl -AuthToken $AuthToken `
+            -IpamFQName @("default-domain", "default-project", "default-network-ipam") `
+            -DNSMode 'none'
+    }
+
+    It 'sets DNS mode to default' {
+        Set-IpamDNSMode -ContrailUrl $ContrailUrl -AuthToken $AuthToken `
+            -IpamFQName @("default-domain", "default-project", "default-network-ipam") `
+            -DNSMode 'default-dns-server'
+    }
+
+    It 'sets DNS mode to tenant without DNS servers' {
+        Set-IpamDNSMode -ContrailUrl $ContrailUrl -AuthToken $AuthToken `
+            -IpamFQName @("default-domain", "default-project", "default-network-ipam") `
+            -DNSMode 'tenant-dns-server'
+    }
+
+    It 'sets DNS mode to tenant with DNS servers' {
+        Set-IpamDNSMode -ContrailUrl $ContrailUrl -AuthToken $AuthToken `
+            -IpamFQName @("default-domain", "default-project", "default-network-ipam") `
+            -DNSMode 'tenant-dns-server' -TenantServersIPAddresses @("1.1.1.1", "2.2.2.2")
+    }
+
+    It 'sets DNS mode to virtual' {
```

Comment:
Test names like this one suggest that it always does set something. We should probably name them `It 'can set DNS mode to virtual` and similar in other places.